File: JAVA/Ancestors in Binary Tree.java
class Solution {
    ArrayList<Integer> result = new ArrayList<Integer>();
    public ArrayList<Integer> Ancestors(Node root, int target) {
        solve(root, target);
        return result;
    }
    public boolean solve(Node root, int target){
        if(root==null) return false;
        if(root.data==target)
            return true;
        boolean isAncestor = solve(root.left, target);
        if(!isAncestor)
            isAncestor = solve(root.right, target);
        if(isAncestor)
            result.add(root.data);
        return isAncestor;
    }
}

File: Ancestors in Binary Tree.cpp
class Solution {
  public:
    bool trav(struct Node *root, int target,vector<int>&ans)
    {
        if(root==NULL)
        return false;
        
        if(root->data==target)
       {
            return true ;
       }
      bool lt= trav(root->left,target,ans);
       bool rt=trav(root->right,target,ans);
       if(lt||rt)
        {
            ans.push_back(root->data);
        }
        return lt||rt;
        
    }
    vector<int> Ancestors(struct Node *root, int target) {
        vector<int> ans;
        trav(root,target,ans);
        return ans;
    }
};
